src/corona_hakab_model/infection.py

```diff
@@ -52,4 +59,18 @@ def _perform_infection(self) -> List[Agent]:
             agent = self.manager.agents[index]
             new_infected.append(agent)
 
-        return new_infected
+        return new_infected, infection_methods
+
+    def _get_infection_method(self, agent_index, non_zero_columns):
+        probability_by_conn_type = []
+        for connection_type in ConnectionTypes:
+            possible_infectors = non_zero_columns[connection_type][agent_index]
+            conn_infection_probability = sum([self.manager.matrix.get(
```

Comment:
i don't think that summing works here.
i think it should be:
..._probability = 1 - prod ([1 - self.manager.matrix.get(...) for ...])

the possibility to get infected from 3 meetings isn't the sum of the possibility to get infected in each meeting, but is 1 - the possibility to not get infected at all.

Reply:
fixed, now each infection probability is separate